File: app/src/main/java/com/vitaliyhtc/dagger2investigation/presentation/base/BaseFragment.java
package com.vitaliyhtc.dagger2investigation.presentation.base;

import android.content.Context;

import com.arellomobile.mvp.MvpAppCompatFragment;

import dagger.android.support.AndroidSupportInjection;

abstract public class BaseFragment extends MvpAppCompatFragment {

    @Override
    public void onAttach(Context context) {
        AndroidSupportInjection.inject(this);
        super.onAttach(context);
    }
}

File: app/src/main/java/com/vitaliyhtc/dagger2investigation/presentation/base/BaseFragment.kt
package com.vitaliyhtc.dagger2investigation.presentation.base

import android.content.Context
import com.arellomobile.mvp.MvpAppCompatFragment
import dagger.android.support.AndroidSupportInjection

abstract class BaseFragment : MvpAppCompatFragment() {

    override fun onAttach(context: Context?) {
        AndroidSupportInjection.inject(this)
        super.onAttach(context)
    }
}